src/Sdk/Common/Common/VssHttpRetryMessageHandler.cs

```diff
@@ -125,7 +125,10 @@ public VssHttpRetryMessageHandler(VssHttpRetryOptions options)
                 }
                 catch (TimeoutException)
                 {
-                    throw;
+                    // Let's consider runner TimeoutException as 'retry-able'. The lower layer (i.e. VssHttpMessageHandler)
+                    // special-cases TimeoutException from OperationCanceledException (caused by cancellation requests)
+                    // so we won't be accidentally stalling the runner on e.g. a Ctrl-Z operation.
+                    canRetry = true;
```

Comment:
what will happen after we run out of retry attempts?

Reply:
It depends on the nature of the retry. Right now, if the last error is a `TimeoutException` (independently of the earlier retries), the method will return a `null` response. However, if the last error is a `HttpRequestException`, then we keep the existing throwing behavior.